### PR TITLE
RF2 transform command 

### PR DESCRIPTION
**Pull request by zoliMolnar**
This pull request adds a transform command to RF2-D2

### CODE REVIEW

File: src/main/java/com/b2international/rf2/RF2Transform.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Option;
+import picocli.CommandLine.Parameters;
+
+/**
+ * @since 0.3
+ */
+@Command(
+	name = "transform",
+	description = "Apply an expression to every RF2 row Modifies values in specified columns in a given set of RF2 releases/files"
```
Comment (cmark) [inline src/main/java/com/b2international/rf2/RF2Transform.java]: Fix command description!

File: src/main/java/com/b2international/rf2/RF2Transform.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+			rawScript = script;
+		}
+		
+		final Path outputDirectory = Paths.get(outDir);
+		if (!Files.exists(outputDirectory)) {
+			Files.createDirectories(outputDirectory);
+		}
+
+		final RF2Specification defaultSpecification = getRF2Specification();
+		final RF2File sourceFile = defaultSpecification.detect(Paths.get(path));
```
Comment (cmark) [inline src/main/java/com/b2international/rf2/RF2Transform.java]: Report error if the specified path does not exist.

File: src/main/java/com/b2international/rf2/model/RF2ContentFile.java [hunk 3]
```diff
@@ -111,42 +120,99 @@ public void check(RF2IssueAcceptor acceptor) throws IOException {
...
+										maxVersionDate.set(newMaxVersionDate);
+									}
+								}
+							});
+						} catch (IOException e) {
+							throw new RuntimeException(e);
+						}
+					});
+
+			if (matchingSourceFile.get() == null) {
```
Comment (cmark) [inline src/main/java/com/b2international/rf2/model/RF2ContentFile.java]: Use `Optional.isPresent()`.

File: src/main/java/com/b2international/rf2/model/RF2ContentFile.java [hunk 4]
```diff
@@ -313,6 +375,20 @@ protected final String newLine(String[] values) {
 		return header == null ? specification.getHeader() : header;
 	}
 
+	/**
+	 * @return the current RF2 header indexes mapped by the header's name.
+	 */
+	public final Map<String, Integer> getIndexByHeader() {
+		final Map<String, Integer> indexByHeader = Maps.newHashMap();
+		final List<String> header = Arrays.asList(getHeader());
```
Comment (cmark) [inline src/main/java/com/b2international/rf2/model/RF2ContentFile.java]: Unnecessary `Arrays.asList()`.

File: src/main/java/com/b2international/rf2/model/RF2Directory.java [hunk 5]
```diff
@@ -61,4 +62,22 @@ public void create(RF2CreateContext context) throws IOException {
 		Files.createDirectories(getPath());
 	}
 
+	@Override
+	public void transform(RF2TransformContext context) throws IOException {
+		// there's nothing to transform so we'll just create the directory to its new location
+		final RF2File newRF2Directory = getRF2FileName().createRF2File(context.getParent(), context.getSpecification());
+
+		Files.createDirectories(newRF2Directory.getPath());
+		context.log("Creating directory '%s'", getPath());
```
Comment (cmark) [inline src/main/java/com/b2international/rf2/model/RF2Directory.java]: Move this logging statement to the beginning of the method.

File: src/main/java/com/b2international/rf2/model/RF2Release.java [hunk 6]
```diff
@@ -130,7 +131,28 @@ public void create(RF2CreateContext context) throws IOException {
...
 			
 		}
-		context.log().log("Created RF2 release at %s", getPath());
+		context.log("Created RF2 release at %s", getPath());
 	}
-	
+
+	@Override
+	public void transform(RF2TransformContext context) throws IOException {
+		final RF2File release = getRF2FileName().createRF2File(context.getParent(), context.getSpecification());
```
Comment (cmark) [inline src/main/java/com/b2international/rf2/model/RF2Release.java]: Add logging statement here, please.

File: src/main/java/com/b2international/rf2/model/RF2UnrecognizedFile.java [hunk 7]
```diff
@@ -38,9 +40,16 @@ public void visit(Consumer<RF2File> visitor) {
...
 	public void create(RF2CreateContext context) throws IOException {
-		context.log().warn("Creating unrecognized files like '%s' is not supported yet.", getRF2FileName());
+		context.warn("Creating unrecognized files like '%s' is not supported.", getRF2FileName());
 	}
-	
+
+    @Override
+    public void transform(RF2TransformContext context) throws IOException {
+	    // In case of unrecognized files transform is essentially a copy
+		context.log("Copying  unrecognized '%s'...", getPath());
```
Comment (cmark) [inline src/main/java/com/b2international/rf2/model/RF2UnrecognizedFile.java]: Remove double space from log message.

**Comment by zoliMolnar** [comment]
Thank you!

**Pull request merged**